### ISSUE TITLE
StableDiffusionXLInpaintPipeline not working

### ISSUE THREAD

**Issue by adhikjoshi** [huggingface/diffusers#4109]
### Describe the bug

Getting this error in StableDiffusionXLInpaintPipeline,

File "/opt/conda/envs/ldm/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_inpaint.py", line 1084, in __call__
latents_outputs = self.prepare_latents(
^^^^^^^^^^^^^^^^^^^^^
File "/opt/conda/envs/ldm/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_inpaint.py", line 661, in prepare_latents
image_latents = self._encode_vae_image(image=image, generator=generator)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 File "/opt/conda/envs/ldm/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_inpaint.py", line 704, in _encode_vae_image
image_latents = image_latents.to(dtype)
^^^^^
UnboundLocalError: cannot access local variable 'dtype' where it is not associated with a value

### Reproduction

```
vae = AutoencoderKL.from_pretrained("madebyollin/sdxl-vae-fp16-fix", torch_dtype=torch.float16, force_upcast=False)

pipe = DiffusionPipeline.from_pretrained("stabilityai/stable-diffusion-xl-base-0.9", torch_dtype=torch.float16, vae=vae, use_safetensors=True, variant="fp16", use_auth_token=hf_token).to("cuda")

result = StableDiffusionXLInpaintPipeline(**pipe.components)(**args_dict)

```

### System Info

pip3 install git+https://github.com/huggingface/diffusers.git

### Who can help?

@pa

**Comment by adhikjoshi** [comment]
Works well, if we don't use "madebyollin/sdxl-vae-fp16-fix" vae

**Comment by patrickvonplaten** [comment]
Thanks for the issue @adhikjoshi, should be fixed in #4113 

BTW it would be super nice if you could try to make the code snippets fully reproducible in issue. Like it would be amazing if we could just copy paste the code snippet here: https://github.com/huggingface/diffusers/issues/4109#issue-1805921377 into a Python shell and we get the issue. 

This doesn't work here because many things are not defined (like what is `args_dict`?). We're getting lots of issues every day, so it would be incredibly nice to also try to make the code snippets fully reproducible :heart:
